core/src/Revolution/modMenu.php

```diff
@@ -94,6 +94,11 @@ protected function getLanguageMenu()
 
         $this->xpdo->lexicon->load('core:languages');
 
+        $currentPage = '';
+        if (!empty($_SERVER['QUERY_STRING'])) {
```

Comment:
$_SERVER['QUERY_STRING'] could contain more than the page request parameter.

Maybe you could use `$params = $this->modx->request->getParameters` to access the values, add the switch parameter and `str_replace('%2F', '/', $this->modx->toQueryString($params))` as params value. Then a CMP would be open after the switch etc.